Packs/ApiModules/Scripts/ServiceNowApiModule/ServiceNowApiModule.py

```diff
@@ -0,0 +1,166 @@
+            if 'SSL Certificate Verification Failed' in e.args[0]:
+                return_error('SSL Certificate Verification Failed - try selecting \'Trust any certificate\' '
+                             'checkbox in the integration configuration.')
+            raise DemistoException(e.args[0])
+
+    def login(self, username: str, password: str):
+        """
+        Generate a refresh token using the given client credentials and save it in the integration context.
+        """
+        data = {
+            'client_id': self.client_id,
+            'client_secret': self.client_secret,
+            'username': username,
+            'password': password,
+            'grant_type': 'password'
+        }
+        try:
+            headers = {
+                'Content-Type': 'application/x-www-form-urlencoded'
+            }
+            res = super()._http_request(method='POST', url_suffix=OAUTH_URL, resp_type='response', headers=headers,
+                                        data=data)
+            try:
+                res = res.json()
+            except ValueError as exception:
+                raise DemistoException('Failed to parse json object from response: {}'.format(res.content), exception)
+            if 'error' in res:
+                return_error(
+                    f'Error occurred while creating an access token. Please check the Client ID, Client Secret '
+                    f'and that the given username and password are correct.\n{res}')
+            if res.get('refresh_token'):
+                refresh_token = {
+                    'refresh_token': res.get('refresh_token')
+                }
+                set_integration_context(refresh_token)
+        except Exception as e:
+            return_error(f'Login failed. Please check the instance configuration and the given username and password.\n'
+                         f'{e.args[0]}')
+
+    def get_access_token(self):
+        """
+        Get an access token that was previously created if it is still valid, else, generate a new access token from
+        the client id, client secret and refresh token.
+        """
+        previous_token = get_integration_context()
+
+        # Check if there is an existing valid access token
+        if previous_token.get('access_token') and previous_token.get('expiry_time') > date_to_timestamp(datetime.now()):
+            return previous_token.get('access_token')
+        else:
+            data = {'client_id': self.client_id,
+                    'client_secret': self.client_secret}
+
+            # Check if a refresh token exists. If not, raise an exception indicating to call the login function first.
+            if previous_token.get('refresh_token'):
+                data['refresh_token'] = previous_token.get('refresh_token')
+                data['grant_type'] = 'refresh_token'
+            else:
+                raise Exception('Could not create an access token. User might be not logged in. Try running the'
```

Comment:
please add a unit test for this case.

Reply:
Option (c) in the UT tests this case. I updated the documentation of the tests to clarify it.